Nyx92/3D-Model-Ecommerce-Frontend
# feat: 3d model rendering

## Changes
- added fully configurable Clicky 3d model 

## How to test
- `npm start`, and navigate to http://localhost:3000/clicky
- Verify that the model Canvas takes up most of the page
- Click on model parts and verify that the name the current active part is displayed in the top left of the page
- Use the colour picker to change the colours of parts, one at a time. Note that there might be a small lag between dragging the colour picker options and the colour actually rendering on the model.

## Known Issues
- `position: absolute` on the colour picker component breaks colour selection functionality. picker is left as a block for now.
- `modelState.currentItem` does not default to `null` after clicking out of the model, so the colour picker currently does not disappear. 

## Screenshots (if any)
![image](https://user-images.githubusercontent.com/47405262/193096946-dd6ea461-d166-4e11-bd64-45fee070b600.png)
^ Initial State on load

![image](https://user-images.githubusercontent.com/47405262/193097247-a3f98706-a481-4837-a44f-00c22cb0d016.png)
^ Zoom/Orbit demo

![image](https://user-images.githubusercontent.com/47405262/193097695-ae22a2e3-c7ad-4fcb-9309-d11de3fbe652.png)
^ After all parts are customized

## To Do
- POST model state data to DB
